File: rust/src/group_anagrams_49.rs
use crate::Solution;
use std::collections::HashMap;
impl Solution {
    pub fn group_anagrams(strs: Vec<String>) -> Vec<Vec<String>> {
        let mut hm: HashMap<[u8; 26], Vec<String>> = HashMap::with_capacity(strs.len());
        let offset = 'a' as usize;
        for s in strs {
            let mut key: [u8; 26] = [0; 26];
            for char in s.chars() {
                key[char.to_ascii_lowercase() as usize - offset] += 1;
            }
            hm.entry(key)
                .and_modify(|v| v.push(s.clone()))
                .or_insert(vec![s]);
        }
        hm.values().cloned().collect::<Vec<Vec<String>>>()
    }
}

File: python/group_anagrams_49.py
from typing import List
from collections import defaultdict
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        ans = []
        hm = defaultdict(list)
        for word in strs:
          j = ''.join(sorted(word))
          hm[j].append(word)
        for (k,v) in hm.items():
            ans.append(v)
        return ans 
